### PR TITLE
docs: mime type guide

### PR DESCRIPTION
**Pull request by florian-hoenicke**
first draft for the mime type guide

### CODE REVIEW

File: chapters/mime-type/index.rst
Comment (JoanFM) [inline chapters/mime-type/index.rst]: I think is better to use an assert here to showcase the value

File: chapters/mime-type/index.rst
Comment (JoanFM) [inline chapters/mime-type/index.rst]: same here

File: chapters/mime-type/index.rst
Comment (JoanFM) [inline chapters/mime-type/index.rst]: and here

File: chapters/mime-type/index.rst
Comment (JoanFM) [inline chapters/mime-type/index.rst]: It is not so clear if OCR would need to be a Segmenter or a Crafter inmho, so it could be missleading?

File: chapters/mime-type/index.rst
Comment (JoanFM) [inline chapters/mime-type/index.rst]: typo in docments

**Review by JoanFM — DISMISSED** [review]

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: great idea. Using assert is much nicer

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: check

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: Yes. You are right. We first have to see how we would implement OCR.

Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: same with video to image segmentation, It is also doing segmenting and crafting in one run. The video has to be decoded (crafting), split into images (segmenting) and encoded into images (crafting).
I will bring up a discussion on how to design the interface of the crafter and segmenter.

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: check

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: check

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: I removed the ocr example

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: Searching `mime type` on Google returns most pages with `MIME type`. Suggest we use that

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: - Suggest removing `in order` usually. It's generally unnecessary imho
- `Document`

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: `illustrates` -> `shows` (simpler word)

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: Document

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: Drivers - check all component terms for caps

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: `the full control` -> `full control`
`like shown` -> `as shown`

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: `on its own` -> `of its own`

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: Drivers

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: `are creating` -> `create`
`are responsible for assigning` -> `assign` ?
~for instance~

File: chapters/mime-type/index.rst
Comment (alexcg1) [inline chapters/mime-type/index.rst]: `illustrates` -> `shows`

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: chapters/mime-type/index.rst
Comment (FionnD) [inline chapters/mime-type/index.rst]: Please use title case https://titlecase.com/

File: chapters/mime-type/index.rst
Comment (FionnD) [reply chapters/mime-type/index.rst]: You can simplify the language: 

For most use cases, chunks have the same mime type as their parent Document. [I don't think you need an example of this]. 

In some use cases, a different parent and chunk mime type is required. Such as processing video, where the chunk would be images. The ``Segmenter``is responsible for assigning the correct mime type to the chunks when they are created.

File: chapters/mime-type/index.rst
Comment (FionnD) [inline chapters/mime-type/index.rst]: Documents

File: chapters/mime-type/index.rst
Comment (FionnD) [inline chapters/mime-type/index.rst:7]: Please add something like: 

This guide explains what a mime type is, how to assign them manually or automatically to a Document and how you can work with them on a Chunk or Driver level.

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: Thanks - changed it to
`The ``Segmenters`` are able to assign the right MIME type to the Chunks they create.`

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: Nice! Thanks. I didn't know there are special rules for titles

File: chapters/mime-type/index.rst
Comment (JoanFM) [inline chapters/mime-type/index.rst]: Chunks are mostly created by `Segmenters`. A user can create `Chunks` as well.

File: chapters/mime-type/index.rst
Comment (JoanFM) [inline chapters/mime-type/index.rst]: this code snippet is going to be outdated by redesign of Driver. I think if the point is to explain `mime_type`, better to simplify the code to show things about the `mime_type` without the rest of the noise.

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: Thanks a lot!

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst:7]: I added it thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: good point. Thanks

File: chapters/mime-type/index.rst
Comment (florian-hoenicke) [reply chapters/mime-type/index.rst]: Valid point!
Do you have an idea what feature of the mime_type I should promote?
At the moment I could not find any other usage. The driver case was somehow made up by me.

Comment (JoanFM) [reply chapters/mime-type/index.rst]: No, I think u explained everything there is to say very well.

File: chapters/core/guides/index.rst
Comment (FionnD) [inline chapters/core/guides/index.rst:40]: Can you actually just add this a page inside of the chapters folder directly? Not within its own folder? We are trying to flatten our folder structure for better SEO. 

so `chapters/mime_types.rst`

Comment (florian-hoenicke) [reply chapters/core/guides/index.rst:40]: nice idea! having so many index.htmls is not so good. you are right

Comment (florian-hoenicke) [reply chapters/core/guides/index.rst:40]: @FionnD What would you think of changing all file paths in one run instead of changing them one after another?
Therefore, we would avoid inconsistencies.

Comment (FionnD) [reply chapters/core/guides/index.rst:40]: My thoughts exactly! A community member is in the process of doing that in this [PR](https://github.com/jina-ai/docs/pull/98). So I think we should leave it to him and not interfere

**Comment by florian-hoenicke** [comment]
Can it be merged? @alexcg1

**Review by FionnD — APPROVED** [review]

**Review by cristianmtr — APPROVED** [review]

**Comment by alexcg1** [comment]
Do you still need another review from me @FionnD ? You wrote the guide, so I guess you should be the one to merge it.

Says merging is blocked because I requested some changes. I guess you addressed those changes, but did you click the 'resolve' button? (I can't see the changes I requested now)

**Comment by FionnD** [comment]
> Do you still need another review from me @FionnD ? You wrote the guide, so I guess you should be the one to merge it.
> 
> Says merging is blocked because I requested some changes. I guess you addressed those changes, but did you click the 'resolve' button? (I can't see the changes I requested now)

No need to review, please just approve the changes to unblock the merge

**Review by alexcg1 — APPROVED** [review]

**Comment by alexcg1** [comment]
Still need the OK from @JoanFM as well

**Pull request merged**
